From 724446a4ebebc3b951737c6cdf1c68315f4168b3 Mon Sep 17 00:00:00 2001
From: Pierre-Eric Pelloux-Prayer <pierre-eric.pelloux-prayer@amd.com>
Date: Fri, 3 May 2024 18:05:37 +0200
Subject: [PATCH 14/16] drm/amdgpu: remove duplicate command name array

No need to have an additional array since the names are already
part of the ccmd_dispatch struct.
Use this opportunity to fix the size of the statistics buffer.

Part-of: <https://gitlab.freedesktop.org/virgl/virglrenderer/-/merge_requests/1362>
---
 src/drm/amdgpu/amdgpu_renderer.c | 20 ++++----------------
 1 file changed, 4 insertions(+), 16 deletions(-)

diff --git a/src/drm/amdgpu/amdgpu_renderer.c b/src/drm/amdgpu/amdgpu_renderer.c
index b9d0c377..6537275f 100644
--- a/src/drm/amdgpu/amdgpu_renderer.c
+++ b/src/drm/amdgpu/amdgpu_renderer.c
@@ -93,7 +93,7 @@ struct amdgpu_context {
    struct hash_table_u64 *id_to_ctx;
 
    struct {
-      struct cmd_stat s[8];
+      struct cmd_stat s[9];
       uint64_t last_print_ms;
    } statistics;
 
@@ -1124,18 +1124,6 @@ static const struct ccmd {
    HANDLER(SET_PSTATE, set_pstate),
 };
 
-const char * __cmd_names[] = {
-   "QUERY_INFO   ",
-   "GEM_NEW      ",
-   "BO_VA_OP     ",
-   "CS_SUBMIT    ",
-   "SET_METADATA ",
-   "BO_QUERY_INFO",
-   "CREATE_CTX   ",
-   "RESERVE_VMID ",
-   "SET_PSTATE   ",
-};
-
 static int
 submit_cmd_dispatch(struct amdgpu_context *ctx, const struct vdrm_ccmd_req *hdr)
 {
@@ -1154,8 +1142,8 @@ submit_cmd_dispatch(struct amdgpu_context *ctx, const struct vdrm_ccmd_req *hdr)
    }
 
    struct cmd_stat * const stat = &ctx->statistics.s[hdr->cmd - 1];
-   print(2, "command: %s (seqno:%u, sz:%zu)", __cmd_names[hdr->cmd - 1], hdr->seqno,
-         ccmd->size);
+   print(2, "command: %s (seqno: %u, size:%zu)",
+         ccmd->name, hdr->seqno, ccmd->size);
 
    uint64_t start = util_current_thread_get_time_nano();
 
@@ -1205,7 +1193,7 @@ submit_cmd_dispatch(struct amdgpu_context *ctx, const struct vdrm_ccmd_req *hdr)
          if (ctx->statistics.s[i].count) {
             int n = (int)roundf(100 * ctx->statistics.s[i].count / (float)total);
             printf("\t%s %3d%% (n: %*d, min: %.3f ms, max: %.3f ms, avg: %.3f ms)\n",
-                   __cmd_names[i],
+                   ccmd_dispatch[i + 1].name,
                    n,
                    align,
                    ctx->statistics.s[i].count,
-- 
2.17.1

